www/main/desktop_scripts.js

```diff
@@ -127,50 +127,50 @@ module.exports = {
-                      // TODO: Add sttmdesktop.realm.management
-                      const oldDBs = ['data.db', 'sttmdesktop.realm', 'sttmdesktop.realm.lock'];
-                      oldDBs.forEach(oldDB => {
-                        const oldDBPath = path.resolve(userDataPath, oldDB);
-                        fs.access(oldDBPath, err => {
-                          if (!err) {
-                            fs.unlink(oldDBPath, err1 => {
-                              if (err1) {
-                                // eslint-disable-next-line no-console
-                                console.log(`Could not delete old database ${oldDB}: ${err1}`);
-                              }
-                            });
-                          }
+                    ipcRenderer.emit('database-progress', JSON.stringify({ percent: 1 }));
+                    try {
+                      extract(dbCompressed, { dir: newDBFolder }).then(() => {
+                        fs.chmodSync(newDBPath, '755');
+                        // Save the hash for comparison next time
+                        store.set('curDBHash', newestDBHash);
+                        // Delete compressed database
+                        fs.unlinkSync(dbCompressed);
+                        // Replace current DB file with new version
+                        fs.renameSync(newDBPath, dbPath);
+                        if (dbPlatform === 'realm') {
+                          fs.renameSync(newDBSchema, dbSchema);
+                        }
+                        module.exports.initDB();
+                        // Delete old DBs
+                        // TODO: Update to check if directory and use fs.rmdir
+                        // TODO: Add sttmdesktop.realm.management
+                        const oldDBs = ['data.db', 'sttmdesktop.realm', 'sttmdesktop.realm.lock'];
+                        oldDBs.forEach((oldDB) => {
+                          const oldDBPath = path.resolve(userDataPath, oldDB);
+                          fs.access(oldDBPath, (err) => {
+                            if (!err) {
+                              fs.unlink(oldDBPath, (err1) => {
+                                if (err1) {
+                                  // eslint-disable-next-line no-console
+                                  console.log(`Could not delete old database ${oldDB}: ${err1}`);
+                                }
+                              });
+                            }
+                          });
                         });
+                        const win = remote.getCurrentWindow();
+                        win.setProgressBar(-1); 
                       });
-                      const win = remote.getCurrentWindow();
-                      win.setProgressBar(-1);
-                    });
+                    } catch (err) {
+                      // handle any errors
+                      console.log(err0);
+                    }
```

Comment:
Veerji, aren't we gonna show this error on frontend?

Reply:
This is from old code, lets keep it as it is for now and we can have a separate PR for this later.

Reply:
Sure veerji thanks